clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -2602,6 +2602,9 @@ public void testSkippingAbortedTransactions() {
         assertTrue(fetcher.hasCompletedFetches());
 
         Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> fetchedRecords = fetchedRecords();
+        assertTrue(fetchedRecords.get(tp0).isEmpty());
```

Comment:
Instead of nothing, we return an empty list if we step over a tx-marker. After a second `fetchRecords()` we return nothing.